Framework/API/src/DetectorInfo.cpp

```diff
@@ -149,11 +174,17 @@ void DetectorInfo::setRotation(const size_t index,
  * typically still influence detector positions. */
 void DetectorInfo::setPosition(const Geometry::IComponent &comp,
                                const Kernel::V3D &pos) {
-  using namespace Geometry::ComponentHelper;
-  TransformType positionType = Absolute;
-  moveComponent(comp, *m_pmap, pos, positionType);
 
-  if (!dynamic_cast<const Geometry::Detector *>(&comp)) {
+  if (const auto *det = dynamic_cast<const Geometry::Detector *>(&comp)) {
+    const auto index = indexOf(det->getID());
+    m_detectorInfo.setPosition(index, Kernel::toVector3d(pos));
+  } else {
```

Comment:
@OwenArnold This was added such that we are able to update detector positions if a parent is moved (not in this PR, note that the function already existed before). This will become part of `ComponentInfo`, but right now there is no other way to do this.